cognite_toolkit/cdf_tk/run.py

```diff
@@ -177,18 +177,17 @@ def run_local_function(
     print(f"[bold]Building for environment {build_env} using {source_path!s} as sources...[/bold]")
     config.set_environment_variables()
     build_dir = tempfile.mkdtemp(prefix="build.", suffix=".tmp", dir=Path.cwd())
+
     found = False
-    for root_module, root_values in config.modules.items():
-        if not isinstance(root_values, dict):
+    for function_dir in source_path.glob("**/functions"):
+        if not function_dir.is_dir():
             continue
-        for group_module, group_values in root_values.items():
-            if not isinstance(group_values, dict):
-                continue
-            for module_name in group_values:
-                if Path(source_path / f"{root_module}/{group_module}/{module_name}/functions/{external_id}").is_dir():
-                    config.environment.selected_modules_and_packages = [f"{module_name}"]
-                    found = True
-                    break
+        for path in function_dir.iterdir():
+            if path.is_dir() and path.name == external_id:
+                config.environment.selected_modules_and_packages = [path.parts[-3]]
```

Comment:
We have a multiple issues with solutions like this in the build command, were we make assumptions on where the module directory is relative to a folder inside the resource folder. I think we need to have a more robust way of getting module from a path.

Reply:
I written a more robust utility function for getting the module from an arbitrary path:

https://github.com/cognitedata/cdf-project-templates/pull/327